### PR TITLE
Feature/crm/reviews page

### PR DESCRIPTION
**Pull request by sivlvy**
https://app.clickup.com/t/8693kw3tr

### CODE REVIEW

File: src/components/feedbacks/form/FeedbackForm.tsx [hunk 1]
```diff
@@ -15,39 +19,36 @@ const DEFAULT_VALUE = {
...
   const [rating, setRating] = useState(0)
   const [file, setFile] = useState<typeof DEFAULT_VALUE>(DEFAULT_VALUE)
   const [fileSelected, setFileSelected] = useState(false)
   const [isModalOpen, setIsModalOpen] = useState(false)
+  const [text, setText] = useState('')
   const { isLoggedIn } = useAuth()
 
   const [searchParams] = useSearchParams()
 
+  const dispatch: AppDispatch = useDispatch()
```
Comment (pqoqubbw) [inline src/components/feedbacks/form/FeedbackForm.tsx]:
```suggestion
  const dispatch = useDispatch<AppDispatch>()
```

File: src/components/feedbacks/form/FeedbackForm.tsx [hunk 2]
```diff
@@ -15,39 +19,36 @@ const DEFAULT_VALUE = {
...
+
   const searchToken = Object.fromEntries(searchParams.entries())
 
   useEffect(() => {
     Object.keys(searchToken).length > 0 && setIsModalOpen(true)
   }, [searchToken])
+  // const BASE_URL = import.meta.env.VITE_API_URL
```
Comment (pqoqubbw) [inline src/components/feedbacks/form/FeedbackForm.tsx]: remove unused code

File: src/components/feedbacks/form/FeedbackForm.tsx [hunk 3]
```diff
@@ -74,14 +75,21 @@ const FeedbackForm = () => {
     setFileSelected(false)
   }
 
+  const handleNameChange = (e: ChangeEvent<HTMLInputElement>) => [
+    setName(e.target.value)
+  ]
```
Comment (pqoqubbw) [inline src/components/feedbacks/form/FeedbackForm.tsx]:
```suggestion
  const handleNameChange = (e: ChangeEvent<HTMLInputElement>) => {
    setName(e.target.value)
  }
```

Comment (pqoqubbw) [reply src/components/feedbacks/form/FeedbackForm.tsx]: also, you do not need this function, you can just pass `setName`

File: src/pages/crm-reviews-page/CrmReviewsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }
+
+  useEffect(() => {
+    fetchReviews()
+  }, [page])
+
+  const handleStatusChange = () => {
+    fetchReviews()
+  }
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/CrmReviewsPage.tsx]: same here, it is an unnecessary wrapper for the function

File: src/pages/crm-reviews-page/CrmReviewsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  return (
+    <Box p="44px 30px 34px 30px" color="illustrations.darker">
+      <Box>
+        <Typography variant="h3">Відгуки </Typography>
+        <div style={{ height: '100%', width: '100%', paddingTop: '30px' }}>
+          <ReviewsList items={reviews} onStatusChange={handleStatusChange} />
+        </div>
+        <PaginationCRM page={page} pageQty={0} setPage={setPage} />
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/CrmReviewsPage.tsx]:
```
pageQty={0}
```
move it to variable as `PAGE_SIZE`

File: src/pages/crm-reviews-page/ReviewsList/ReviewsList.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import styles from './ReviewsListItem/reviews-list-item.module.scss'
+
+import { Review } from '../../../redux/feedbacks/slice'
+import { ReviewsListItem } from './ReviewsListItem/ReviewsListItem'
+
+export interface ReviewsListProps {
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsList.tsx]: why do you need `interface` here?

File: src/pages/crm-reviews-page/ReviewsList/ReviewsList.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          <ReviewsListItem
+            id={id}
+            key={id}
+            rating={rating}
+            description={description}
+            time={created_at}
+            name={first_name}
+            isActive={is_deleted}
+            images={images}
+            onStatusChange={onStatusChange}
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsList.tsx]:
```suggestion
          id,
          is_deleted,
          user: { first_name },
          created_at,
          ...rest
        }: Review) => (
          <ReviewsListItem
            id={id}
            key={id}
            time={created_at}
            name={first_name}
            isActive={is_deleted}
            onStatusChange={onStatusChange}
            {...rest}
```

File: src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+import IconButton from '@mui/material/IconButton'
+import DeleteIcon from '../../../../icons/delete.svg?react'
+import Tooltip from '@mui/material/Tooltip'
+import ChangeIcon from '../../../../icons/crm-file.svg?react'
+import { Images } from '../../../../redux/feedbacks/slice'
+import { ModalCustom } from '../../../../components/Modal-custom-btn/ModalCustomWindow'
+import { useState } from 'react'
+import axiosInstance from '../../../../axios/settings'
+
+export interface ReviewsListItemProps {
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: same here

Comment (pqoqubbw) [reply src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: and check be types, should be already created. use types from be and do not create your own

File: src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  const handleAddToArchive = () => {
+    const fetchAdd = async () => {
+      try {
+        await axiosInstance.put('/api/reviews/archive', { id })
+
+        onStatusChange()
+      } catch (err) {
+        console.error(err)
+      }
+    }
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: why do you need function inside another function? why not to just call it?

File: src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+  const handleRemoveFromArchive = () => {
+    const fetchRemove = async () => {
+      try {
+        await axiosInstance.put('/api/reviews/unarchive', { id })
+
+        onStatusChange()
+      } catch (err) {
+        console.error(err)
+      }
+    }
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: same here

File: src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    fetchRemove()
+  }
+  return (
+    <li className={styles.item}>
```
Comment (pqoqubbw) [inline src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: if i am not mistaken, it is just flex list of items, so, why do you use `li`?

File: src/redux/feedbacks/operations.ts [hunk 12]
```diff
@@ -26,6 +26,7 @@ export const fetchReviews = createAsyncThunk<Review[], FetchReviewsParams>(
 export const submitReview = createAsyncThunk(
   'api/reviews/create',
   async (formData, { rejectWithValue }) => {
+    console.log(formData)
```
Comment (pqoqubbw) [inline src/redux/feedbacks/operations.ts]: remove console.log

File: src/redux/feedbacks/slice.ts
Comment (pqoqubbw) [inline src/redux/feedbacks/slice.ts:1]: check be types

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/pages/crm-reviews-page/CrmReviewsPage.tsx (hunk 4 above)
Comment (sivlvy) [reply src/pages/crm-reviews-page/CrmReviewsPage.tsx]: in ReviewListItem, I need to update all reviews when the status changes, so I pass the function like this. Is there any other way to do it?

File: src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx (hunk 11 above)
Comment (sivlvy) [reply src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: Should I use 'div' or something from mui?

Comment (pqoqubbw) [reply src/pages/crm-reviews-page/ReviewsList/ReviewsListItem/ReviewsListItem.tsx]: yeah, just div
use lists tags only when you have actual list

File: src/pages/crm-reviews-page/CrmReviewsPage.tsx (hunk 4 above)
Comment (pqoqubbw) [reply src/pages/crm-reviews-page/CrmReviewsPage.tsx]:
```
<Component onUpdate={() => fetchReviews()} />
```
or
```
<Component onUpdate={fetchReviews)} />
```

File: src/redux/feedbacks/slice.ts [hunk 13]
```diff
@@ -53,9 +63,11 @@ export const reviewsSlice = createSlice({
     },
     [fetchReviews.rejected]: handleRejected,
     [submitReview.pending]: handlePending,
-    [submitReview.fulfilled](state) {
+    [submitReview.fulfilled](state, { payload }) {
       state.isLoading = false
       state.error = null
+      console.log(payload)
```
Comment (pqoqubbw) [inline src/redux/feedbacks/slice.ts]: remove console.log

**Review by pqoqubbw — CHANGES_REQUESTED** [review]
fix comments with types, please

**Comment by sivlvy** [comment]
&gt; fix comments with types, please
 I changed the code using types from the types/models directory, or am I misunderstanding the point?

**Review by pqoqubbw — APPROVED** [review]

**Pull request merged**